### PR TITLE
Fixing _get_standard_colors to honor num_colors arguments (#20585)

### LINKED ISSUE #20585: Default behavior of plot.bar changed to plot a different color for each bar
#### Code Sample, a copy-pastable example if possible
**BEFORE**
```python
df_india.groupby('sector').id.count().sort_values().plot.bar();
```
would result in a bar plot with all bars having the same color

**AFTER** version 0.20
```python
df_india.groupby('sector').id.count().sort_values().plot.bar(color='cornflowerblue');
```
I need to explicitly provide a color, else each bar is colored differently

Minimal reproducible code:

```
import pandas as pd
df = pd.DataFrame({'account-start': ['2017-02-03', '2017-03-03', '2017-01-01'],
                   'client': ['Alice Anders', 'Bob Baker', 'Charlie Chaplin'],
                   'balance': [-1432.32, 10.43, 30000.00],
                   'db-id': [1234, 2424, 251],
                   'proxy-id': [525, 1525, 2542],
                   'rank': [52, 525, 32],
                   })
df.client.value_counts().plot.bar()
```

Output:

download

#### Problem description
Why was the behavior of Series.plot.bar changed to plot bars with different color? Visually these colors add nothing to the plot as different colors should only be used when they correspond to differences of meaning in the data.

Why is the default behavior to provide an unnecessarily visually overwhelming graph? It took me some time to realize why my bars suddenly started acting strangely. Now, I pass color='cornflowerblue' to get all my bars the same pleasant hue.

Reference for visual appeal and the use of colors: http://www.perceptualedge.com/articles/visual_business_intelligence/rules_for_using_color.pdf

Would it be possible to revert to the pre-0.20 behavior?

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.4.final.0
python-bits: 64
OS: Linux
OS-release: 4.9.0-5-amd64
machine: x86_64
processor: 
byteorder: little
LC_ALL: C.UTF-8
LANG: C.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.22.0
pytest: 3.2.1
pip: 9.0.1
setuptools: 38.5.2
Cython: 0.26.1
numpy: 1.14.2
scipy: 1.0.0
pyarrow: 0.8.0
xarray: 0.10.1
IPython: 6.1.0
sphinx: 1.6.3
patsy: 0.4.1
dateutil: 2.6.0
pytz: 2018.3
blosc: None
bottleneck: 1.2.1
tables: 3.4.2
numexpr: 2.6.2
feather: 0.4.0
matplotlib: 2.1.0
openpyxl: 2.4.8
xlrd: 1.1.0
xlwt: 1.3.0
xlsxwriter: 1.0.2
lxml: 4.1.0
bs4: 4.6.0
html5lib: 0.9999999
sqlalchemy: 1.1.13
pymysql: None
psycopg2: None
jinja2: 2.9.6
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: 0.6.0
</details>

### PR DESCRIPTION
**Pull request by atulagrwl**
- [ ] closes #20585
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/plotting/test_series.py [hunk 1]
```diff
@@ -268,6 +269,34 @@ def test_bar_user_colors(self):
                     (1., 0., 0., 1.)]
         assert result == expected
 
+    def test_bar_user_colors_limited(self):
+        s = Series([1, 2, 3, 4])
```
Comment (TomAugspurger) [inline pandas/tests/plotting/test_series.py]: Add a reference to the original issue in a comment here.

**Review by TomAugspurger — COMMENTED** [review]
Looks good at a glance.

Could you add a release note to 0.24.0 in the bugfix section?

**Comment by atulagrwl** [comment]
@TomAugspurger - Thanks for quick review. Would really appreciate merging it to master soon (this would be my first contribution to get into master).

File: pandas/plotting/_core.py [hunk 2]
```diff
@@ -1233,6 +1233,16 @@ def _plot(cls, ax, x, y, w, start=0, log=False, **kwds):
     def _start_base(self):
         return self.bottom
 
+    def _get_colors(self, num_colors=None, color_kwds='color'):
+        color = self.kwds.get(color_kwds)
```
Comment (jreback) [inline pandas/plotting/_core.py]: can you add a doc-string here

File: pandas/plotting/_style.py [hunk 3]
```diff
@@ -91,7 +91,9 @@ def _maybe_valid_colors(colors):
             # mpl will raise error any of them is invalid
             pass
 
-    if len(colors) != num_colors:
```
Comment (jreback) [inline pandas/plotting/_style.py]: can you add a comment here

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
tests are failing as well

**Comment by dyoung418** [comment]
@atulagrwl do you plan on making the changes requested of you above?  I'd love to see this bugfix get merged.

**Comment by WillAyd** [comment]
Closing as stale. Ping if you'd like to pick back up

**Pull request closed without merging**